### DISCUSSION TITLE
CrazyFie 2.1 with 2x LHS2.0 base stations. Takeoff unstable

### DISCUSSION THREAD

**Discussion by AkashJohnSubash** [bitcraze/discussions#544] [Q&A] [answered]
After a couple of crashes on padded floors/ nets (using a off-board trajectory algorithm),  executing take-off is unstable with cfclient. (was stable before crash)
After replacing the propellers & motor corresponding to failing tests, I attempted the take-off again. 
However, after a few seconds of wild swerves, it crashes.

Now the startup-self tests  and propeller tests pass, but the battery tests fail with a small margin.
I checked with second fresh lighthouse deck, but the instability persists.
I would appreciate some advice on where to begin debugging.

Below is a full log:

ESTKALMAN: State out of bounds, resetting
SYS: ----------------------------
SYS: Crazyflie 2.1 is up and running!
SYS: Build 0:b8d2e0b9e983 (2022.12) CLEAN
SYS: I am 0x2039343053335007002E0028 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 1 deck(s) found
DECK_CORE: Calling INIT on driver bcLighthouse4 for deck 0
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Kalman (2) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2022.12
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 19184 bytes
LHFL: Lighthouse bootloader version: 2
LHFL: Bitstream CRC32: 112BC794 [PASS]
LHFL: Firmware version 6 verified, booting deck!
STAB: Ready to fly.
ESTKALMAN: State out of bounds, resetting
HEALTH: Acc noise floor variance X+Y:0.003630, (Z:0.001815)
HEALTH: Motor M1 variance X+Y: 1.04 (Z:0.23), voltage sag:0.31
HEALTH: Motor M2 variance X+Y: 1.85 (Z:3.20), voltage sag:0.31
HEALTH: Motor M3 variance X+Y: 1.90 (Z:0.25), voltage sag:0.27
HEALTH: Motor M4 variance X+Y: 1.48 (Z:1.79), voltage sag:0.31
HEALTH: Battery sag during load test (0.98 > 0.95) [FAILED].
STAB: WARNING: stabilizer loop rate is off (259)
ESTKALMAN: WARNING: Kalman prediction rate low (94)

**Comment by krichardsson** [comment, marked as answer]
The log looks OK. As you say, the battery, or possibly the battery connector is maybe a bit tired and this might cause problems if you take off fast. The lighthouse deck is sensitive to low voltage and if the battery connector is a bit oxidized this could cause a low voltage in the Crazyflie when you spin up the motors a lot. If the lighthouse deck shuts down you could get this result possibly. 
You could try to clean the battery connector with some wd40 to get a better connection. 

Have you verified (by using the client) that you get a reasonable position estimate if you carry the Crazyflie around? 
Is it possible that one of the base stations has moved a bit and you need to redo the lighthouse geometry?
Have you tried to fly using one of the example scripts, for instance [autonomous_sequence_high_level.py](https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/autonomy/autonomous_sequence_high_level.py)?

**Comment by AkashJohnSubash** [comment]
Hi Kristoffer, 

Thank you for taking time to answer, cleaning the connector, changing the motors and redoing the lighthouse geometry fixed it. 
It was most likely an issue with the battery connector, since the lighthouse base stations were working fine for my teammates' drones.
Will provide more detailed information/ steps, if it occurs again.
